### PR TITLE
Add createSpreadsheet

### PR DESCRIPTION
**Pull request by piecubed**
Tested this, seems to work.  Not sure if this is me doing something wrong, but it doesn't work with any spaces in the title.

### CODE REVIEW

**Comment by a-marenkov** [comment]
Hi! Thanks for contributing to gsheets.

I checked title with space and it works, here is the snippet:

```
void main() async {
  final gsheets = GSheets(_credentials);
  final ss = await gsheets.createSpreadsheet('test / test');
  print(ss.id);
  print(await ss.share('dev.a.marenkov@gmail.com', role: PermRole.owner));
}
```

Does it throw the exception when you use space?

**Comment by a-marenkov** [comment]
Also, i think it might be handy to add ability to create worksheets along with creating spreadsheet. What do you think?

File: lib/src/gsheets.dart [hunk 1]
```diff
@@ -65,6 +65,55 @@ class GSheets {
...
+  /// https://developers.google.com/sheets/api/reference/rest/v4/ValueRenderOption
+  ///
+  /// [input] - determines how input data should be interpreted.
+  /// https://developers.google.com/sheets/api/reference/rest/v4/ValueInputOption
+  ///
+  /// Throws Exception if [GSheets]'s scopes does not include SpreadsheetsScope.
+  /// Throws GSheetsException if does not have permission.
+  Future<Spreadsheet> createSpreadsheet(
+      String title, {
+      ValueRenderOption render = ValueRenderOption.unformatted_value,
```
Comment (a-marenkov) [inline lib/src/gsheets.dart:82]:
```suggestion
    List<String> worksheetTitles = const <String>['Sheet1'],
    ValueRenderOption render = ValueRenderOption.unformatted_value,
```

File: lib/src/gsheets.dart [hunk 2]
```diff
@@ -65,6 +65,55 @@ class GSheets {
...
+      ValueInputOption input = ValueInputOption.user_entered
+    }) async {
+    final client = await this.client.catchError((_) {
```
Comment (a-marenkov) [inline lib/src/gsheets.dart:85]:
```suggestion
    except(
      isNullOrEmpty(worksheetTitles),
      'Invalid worksheetTitles ($worksheetTitles)',
    );
    final client = await this.client.catchError((_) {
```

File: lib/src/gsheets.dart [hunk 3]
```diff
@@ -65,6 +65,55 @@ class GSheets {
...
+      // retry once on error
+      _client = null;
+      return this.client;
+    });
+    final response = await client.post(_sheetsEndpoint, body: jsonEncode({
```
Comment (a-marenkov) [inline lib/src/gsheets.dart:90]:
```suggestion
    final worksheets = worksheetTitles
        .map((title) => {
              'properties': {
                'title': title,
                'sheetType': 'GRID',
                'gridProperties': {
                  'rowCount': 1000,
                  'columnCount': 1000,
                }
              },
            })
        .toList();
    final response = await client.post(_sheetsEndpoint, body: jsonEncode({
```

File: lib/src/gsheets.dart [hunk 4]
```diff
@@ -65,6 +65,55 @@ class GSheets {
...
+      'properties': {
+        'title': title,
+      },
```
Comment (a-marenkov) [inline lib/src/gsheets.dart:93]:
```suggestion
      },
      'sheets': worksheets,
```

**Comment by piecubed** [comment]
Just tried it with a space again, and seems like its working.  The error didn't actually say anything about a space, but it was giving me a 500 error code for a while, and I tried removing the space, and it worked.  Not sure what was causing it.

**Comment by piecubed** [comment]
Added all your edits.  I plan on trying to make full API coverage, so be ready for more PRs!

**Pull request merged**